usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlockTest.java

```diff
@@ -0,0 +1,1523 @@
+        // | 8      | 13:08       | 13:50         | 106    | 15            |
+
+        // trips.txt + stop_times.txt
+        // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+        // |---------|--------|-----------|---------|-----------------|----------------|
+        // | 0       | 2      | a         | 7       | 07:00           | 10:00          |
+        // | 0       | 5      | b         | 7       | 17:23           | 18:57          |
+        // | 0       | 8      | c         | 7       | 09:40           | 13:50          |
+
+        // calendar.txt
+        // | serviceId | monday | tuesday | wednesday | thursday | friday | saturday | sunday | start_date | end_date |
+        // |-----------|--------|---------|-----------|----------|--------|----------|--------|------------|----------|
+        // | a         | 1      | 1       | 0         | 0        | 0      | 0        | 0      | 20200801   | 20200831 |
+        // | b         | 1      | 0       | 1         | 0        | 0      | 0        | 0      | 20200901   | 20209030 |
+        // | c         | 0      | 1       | 0         | 1        | 1      | 1        | 0      | 20200801   | 20200831 |
+        //
+        // Here, services a and c overlap: they share the same date range and are both active on tuesday. Specifically
+        // on tuesday, service a starts at 07:00 and ends at 10:00 - and service b starts at 09:40 and ends at 13:30.
+        // Which means that trips 2 and 8 overlap on tuesdays between 09:40 and 10:00: that is an error and should
+        // generate a notice.
+
+        // calendar_dates.txt: file not provided
+        final GtfsDataRepository mockDataRepo = mock(GtfsDataRepository.class);
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final Logger mockLogger = mock(Logger.class);
+        final TimeUtils mockTimeUtils = mock(TimeUtils.class);
+        when(mockTimeUtils.convertIntegerToHMMSS(700)).thenReturn("700");
+        when(mockTimeUtils.convertIntegerToHMMSS(1000)).thenReturn("1000");
+        when(mockTimeUtils.convertIntegerToHMMSS(940)).thenReturn("940");
+        when(mockTimeUtils.convertIntegerToHMMSS(1350)).thenReturn("1350");
+        when(mockTimeUtils.arePeriodsOverlapping(700, 1000,
+                940, 1350)).thenReturn(true);
+
+        final Trip firstMockTrip = mock(Trip.class);
+        when(firstMockTrip.getTripId()).thenReturn("2");
+        when(firstMockTrip.getServiceId()).thenReturn("a");
+        when(firstMockTrip.getBlockId()).thenReturn("7");
+        when(firstMockTrip.hasSameServiceId(any())).thenReturn(false);
+
+
+        final Trip secondMockTrip = mock(Trip.class);
+        when(secondMockTrip.getTripId()).thenReturn("5");
+        when(secondMockTrip.getServiceId()).thenReturn("b");
+        when(secondMockTrip.getBlockId()).thenReturn("7");
+        when(secondMockTrip.hasSameServiceId(any())).thenReturn(false);
```

Comment:
Same on not mocking domain class methods

Reply:
As discussed, implementation of domain layer classes is tested separately, so that tests of use case layer only verify interactions with mocked objects.